Source/Urho3D/CSharp/Managed/Container/SharedPtr.cs

```diff
@@ -0,0 +1,146 @@
+        public SharedPtr(SharedPtr<T> other) : this(other?._value)
+        {
+        }
+
+        /// <summary>Indicates whether the current object is equal to another object of the same type.</summary>
+        /// <param name="other">An object to compare with this object.</param>
+        /// <returns>true if the current object is equal to the <paramref name="other">other</paramref> parameter; otherwise, false.</returns>
+        public bool Equals(SharedPtr<T> other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+            return EqualityComparer<T>.Default.Equals(_value, other._value);
+        }
+
+        /// <summary>Determines whether the specified object is equal to the current object.</summary>
+        /// <param name="obj">The object to compare with the current object.</param>
+        /// <returns>true if the specified object  is equal to the current object; otherwise, false.</returns>
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != this.GetType()) return false;
+            return Equals((SharedPtr<T>)obj);
+        }
+
+        /// <summary>Serves as the default hash function.</summary>
+        /// <returns>A hash code for the current object.</returns>
+        public override int GetHashCode()
+        {
+            return EqualityComparer<T>.Default.GetHashCode(_value);
+        }
+
+        /// <summary>
+        /// Implicitly convert SharedPtr into value.
+        /// </summary>
+        /// <param name="other"></param>
+        public static implicit operator T(SharedPtr<T> @this) => @this._value;
+
+        /// <summary>Indicates whether the two objects are equal.</summary>
+        public static bool operator ==(SharedPtr<T> left, SharedPtr<T> right)
+        {
+            return Equals(left, right);
+        }
+
+        /// <summary>Indicates whether the two objects are not equal.</summary>
+        public static bool operator !=(SharedPtr<T> left, SharedPtr<T> right)
+        {
+            return !Equals(left, right);
+        }
+
+        /// <summary>
+        /// Get the underlying value.
+        /// </summary>
+        public T Ptr => _value;
+
+        /// <summary>Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.</summary>
+        public void Dispose()
+        {
+            if (_value != null)
+            {
+                _value.ReleaseRef();
+            }
+        }
+
+        /// <summary>Returns a string that represents the current object.</summary>
+        /// <returns>A string that represents the current object.</returns>
+        public override string ToString()
+        {
+            return $"{_value}";
+        }
+
+        private readonly T _value;
```

Comment:
In the spirit of consistency i would also rename it to _ptr (this is what C++ `SharedPtr<>` names corresponding field too).